libbeat/cmd/instance/beat.go

```diff
@@ -1066,3 +1069,17 @@ func setUmaskWithSettings(settings Settings) error {
 	}
 	return setUmask(0027) // 0640 for files | 0750 for dirs
 }
+
+type fleetConfigManager interface {
+	ConfigChan() chan<- map[string]interface{}
+}
+
+func startGrpcServer(b *Beat) {
+	if cm, ok := b.ConfigManager.(fleetConfigManager); ok {
+		logp.Info("initiating fleet config manager")
+		s := NewConfigServer(cm.ConfigChan())
+		if err := server.NewGrpcServer(os.Stdin, s); err != nil {
+			panic(err)
+		}
+	}
+}
```

Comment:
Could this initialization logic be only be present in the x-pack/libbeat directory?

Reply:
yes but this would mean we will support only x-pack beats right

Reply:
Yeah, at the moment we should be that way. We can always extend it later for opensource/community beat.